platform/solr/solr-factory/src/main/java/org/codice/solr/client/solrj/SolrClient.java

```diff
@@ -82,6 +89,7 @@
    * @return <code>false</code> if the Solr server this client is connected or the corresponding
    *     core is not available; <code>true</code> otherwise
    * @throws InterruptedException if the current thread was interrupted while waiting
+   * @throws IllegalArgumentException if <code>unit</code> is <code>null</code>
    */
   public boolean isAvailable(long timeout, TimeUnit unit) throws InterruptedException;
```

Comment:
Even the `whenAvailable()` suffers the same fate since by the time, the adapter decides to call your code, the state might have already changed.
At the minimum, when `isAvailable()` returns `false`, you know it is down so you might not want to attempt your operation. If it returns `true`, you can try but it might also fail right away. 

However, because normally, clients should be up more often than they are down, the check does become somewhat useful.